deployment/keystone/types.go

```diff
@@ -321,70 +215,62 @@ func mapDonsToCaps(dons []DonInfo) map[string][]kcr.CapabilitiesRegistryCapabili
 
 // mapDonsToNodes returns a map of don name to simplified representation of their nodes
 // all nodes must have evm config and ocr3 capability nodes are must also have an aptos chain config
-func mapDonsToNodes(dons []DonInfo, excludeBootstraps bool, registryChainSel uint64) (map[string][]*ocr2Node, error) {
-	donToOcr2Nodes := make(map[string][]*ocr2Node)
+func mapDonsToNodes(dons []DonInfo, excludeBootstraps bool, registryChainSel uint64) (map[string][]deployment.Node, error) {
+	donToNodes := make(map[string][]deployment.Node)
 	// get the nodes for each don from the offchain client, get ocr2 config from one of the chain configs for the node b/c
-	// they are equivalent, and transform to ocr2node representation
+	// they are equivalent
 
 	for _, don := range dons {
 		for _, node := range don.Nodes {
-			ocr2n, err := newOcr2NodeFromJD(&node, registryChainSel)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create ocr2 node for node %s: %w", node.ID, err)
-			}
-			if excludeBootstraps && ocr2n.IsBoostrap {
+			if excludeBootstraps && node.IsBootstrap {
 				continue
 			}
-			if _, ok := donToOcr2Nodes[don.Name]; !ok {
-				donToOcr2Nodes[don.Name] = make([]*ocr2Node, 0)
+			if _, ok := donToNodes[don.Name]; !ok {
+				donToNodes[don.Name] = make([]deployment.Node, 0)
 			}
-			donToOcr2Nodes[don.Name] = append(donToOcr2Nodes[don.Name], ocr2n)
-		}
-	}
-
-	return donToOcr2Nodes, nil
-}
-
-func firstChainConfigByType(ccfgs []*v1.ChainConfig, t v1.ChainType) (*v1.ChainConfig, bool) {
-	for _, c := range ccfgs {
-		if c.Chain.Type == t {
-			return c, true
+			donToNodes[don.Name] = append(donToNodes[don.Name], node)
 		}
 	}
-	return nil, false
-}
 
-func registryChainConfig(ccfgs []*v1.ChainConfig, t v1.ChainType, sel uint64) (*v1.ChainConfig, error) {
-	chainId, err := chainsel.ChainIdFromSelector(sel)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get chain id from selector %d: %w", sel, err)
-	}
-	chainIdStr := strconv.FormatUint(chainId, 10)
-	for _, c := range ccfgs {
-		if c.Chain.Type == t && c.Chain.Id == chainIdStr {
-			return c, nil
-		}
-	}
-	return nil, fmt.Errorf("no chain config for chain %d", chainId)
+	return donToNodes, nil
 }
 
 // RegisteredDon is a representation of a don that exists in the in the capabilities registry all with the enriched node data
 type RegisteredDon struct {
 	Name  string
 	Info  capabilities_registry.CapabilitiesRegistryDONInfo
-	Nodes []*ocr2Node
+	Nodes []deployment.Node
 }
 
-func (d RegisteredDon) signers() []common.Address {
+func (d RegisteredDon) signers(chainFamily string) []common.Address {
```

Comment:
would this be easier if passing the selector rather than the family? all the nodes have to have the registry chain enabled for the system to work so should be safe to pass the registry selector

Reply:
add a small test would give confidence of this func

Reply:
this was actually the bug: I used the registry selector but the write nodes didn't have the registry chain defined, so the lookup was failing